benchmarks/yamauchi_takei_2016_anelasticity/anelasticity_temperature.h

```diff
@@ -0,0 +1,136 @@
+/*
+  Copyright (C) 2016 - 2019 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef _aspect_initial_temperature_anelasticity_temperature_h
+#define _aspect_initial_temperature_anelasticity_temperature_h
+
+#include <aspect/initial_temperature/interface.h>
+#include <aspect/utilities.h>
+#include <aspect/simulator_access.h>
+#include <cmath>
+#include <algorithm>
+#include <functional>
+
+
+namespace aspect
+{
+  namespace InitialTemperature
+  {
+    using namespace dealii;
+
+    /**
+     * A class that implements a prescribed temperature field determined from
+     * a AsciiDataProfile input file.
+     *
+     * @ingroup InitialTemperatures
+     */
+    template <int dim>
+    class AnelasticVs2T : public Utilities::AsciiDataInitial<dim>, public Interface<dim>
+    {
+      public:
+	   /**
+	   * Constructor. Initialize variables.
+	   */
+	   AnelasticVs2T ();
+
+       /**
+	   * Initialization function. This function is called once at the
+	   * beginning of the program. Checks preconditions.
+	   */
+	   void
+	   initialize ();
+
+       // Avoid -Woverloaded-virtual:
+	   using Utilities::AsciiDataInitial<dim>::initialize;
+
+       /**
+       * Return the boundary temperature as a function of position. For the
+       * current class, this function returns value from the text files.
+       */
+       double
+       initial_temperature (const Point<dim> &position) const;
+
+       /**
+       * Brent minimization function to find temperature from Vs and isothermal compression from pressure
+       */
+       static
+	   double
```

Comment:
this function has little to do with your class here. Can you move it into the .cc file as a static function?